### DISCUSSION TITLE
Edit entities "friendly name"

### DISCUSSION THREAD

**Discussion by Geek-MD** [frenck/spook#854] [Ideas] [answered]
The idea is to allow edition of entities "friendly name". I have some smart sockets and I want to rename it's friendly name according to seasons, in winter an fall I want it to be named "heater", and in spring and summer I want it to be named "fan".

**Comment by ConnorThomp** [comment]
Yes! I am so desperate for this feature. Being able to use a command that could be integrated into automations and scripts would be amazing.

**Reply by michal-bartak** [reply]
I'm asking for that too.
Currently, if the entity is a member of a device, its friendly name is like "device name entity name".
It's impractical for many reasons depending on integration.
Mainly
* we end up with very long friendly names, mostly not fitting GUI, or
* prefixes might be meaningless if the device is named after mac caddress or another technical information.

MQTT integration allows to overriding entity id by using object_id, also together with the name. But it leads to one of two situations
* The friendly name contains device name and probably part of device name, ie `Shelly 3EM Mainhouse Shelly 3EM Power`
* The friendly name is very indistinct like "phase 1 power". 

Both cases are unacceptable on GUIs.

An automated way to rename entity-friendly names is more than needed for maintaining serious installations

PS for reference, WTH discussion on HA forum: https://community.home-assistant.io/t/wth-no-entity-and-friendly-names-freedom-when-creating-mqtt-entities/818829/28

**Comment by FragMenthor** [comment]
Already asked here: https://github.com/frenck/spook/discussions/573#discussion-6122286

**Comment by teskanoo** [comment, marked as answer]
I think this covers the requested functionality: https://spook.boo/entities#rename-an-entity
